File: cli/template/packages/ruby/lib/{{name}}_do.rb
# frozen_string_literal: true

require_relative '{{name}}_do/version'
require_relative '{{name}}_do/client'

# {{Name}}.do SDK
#
# {{description}}
#
# @example
#   require '{{name}}.do'
#
#   client = {{Name}}Do::Client.new(api_key: ENV['DOTDO_KEY'])
#   rpc = client.connect
#   result = rpc.example
module {{Name}}Do
  class Error < StandardError; end
  class ConnectionError < Error; end
  class ConfigurationError < Error; end
end

File: cli/template/packages/dart/lib/{{name}}_do.dart
/// {{Name}}.do SDK for Dart
///
/// {{description}}
///
/// Example:
/// ```dart
/// import 'package:{{name}}_do/{{name}}_do.dart';
///
/// void main() async {
///   final client = {{Name}}Client(apiKey: Platform.environment['DOTDO_KEY']);
///   await client.connect();
///   // Use the client...
///   await client.disconnect();
/// }
/// ```
library {{name}}_do;

export 'src/client.dart';
export 'src/exceptions.dart';
